interfaces/kmod/kmod.go

```diff
@@ -18,3 +18,61 @@
  */
 
 package kmod
+
+import (
+	"bytes"
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+
+	"github.com/snapcore/snapd/dirs"
+	"github.com/snapcore/snapd/interfaces"
+	"github.com/snapcore/snapd/osutil"
+	"github.com/snapcore/snapd/snap"
+)
+
+func moduleFilePath(snapName string) string {
+	confFileName := fmt.Sprintf("%s.conf", snap.SecurityTag(snapName))
+	return filepath.Join(dirs.SnapKModModulesDir, confFileName)
+}
+
+// syncModulesFile creates a .conf file with list of modules for given snap, or removes it
+// if no modules are requested.
+func syncModulesFile(modules [][]byte, snapName string) (error, bool) {
+	dir := dirs.SnapKModModulesDir
+	if err := os.MkdirAll(dir, 0755); err != nil {
+		return fmt.Errorf("cannot create directory for kmod files %q: %s", dir, err), false
+	}
+
+	glob := interfaces.SecurityTagGlob(snapName)
```

Comment:
Can you have a look at how all the other backends work and how they split between "compute the desired state" and "make the fs consistent with the desired state" please? Specifically the `combineSnippets()` method.  I'd like to stay consistent as this just makes it easier to navigate the code.